src/libraries/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -40,12 +40,15 @@ private static bool IsBusyboxPing(string? pingBinary)
         {
             if (pingBinary != null)
             {
-                string? linkedName = Interop.Sys.ReadLink(pingBinary);
-
-                // If pingBinary is not link linkedName will be null
-                if (linkedName != null && linkedName.EndsWith("busybox", StringComparison.Ordinal))
+                FileInfo info = new FileInfo(pingBinary);
+                // This is Windows way how to say it is symlink
+                if (info?.Attributes.HasFlag(FileAttributes.ReparsePoint) == true)
```

Comment:
I think there is no guarantee that this flag is set on Unix. It should work if we remove this condition. The nullcheck on line 48 will handle it; `ResolveLinkTarget` returns null when [path is not a link](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.resolvelinktarget?view=net-7.0#returns).

Reply:
The linked issue (and previous comment) had concern about cases when it would return `null` for other reasons. So I triode to add more checks. maybe this is all now handled by the public method but I didn't do any extensive testing.

Reply:
Alright, we should be fine. Looks like flag is also set on Unix as well:
https://github.com/dotnet/runtime/blob/b8584841ffed04ec8559629d0392eee9879f5dcb/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L206-L207
Maybe change the comment to something like:

```diff
- // This is Windows way how to say it is symlink
+ // Workaround until there is a better API to determine symlink: https://github.com/dotnet/runtime/issues/53577
```

Reply:
Alright, we should be fine. Looks like flag is set on Unix as well:
https://github.com/dotnet/runtime/blob/b8584841ffed04ec8559629d0392eee9879f5dcb/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L206-L207
Maybe change the comment to something like:

```diff
- // This is Windows way how to say it is symlink
+ // Workaround until there is a better API to determine symlink: https://github.com/dotnet/runtime/issues/53577
```

Reply:
> This is all now handled by the public method but I didn't do any extensive testing.

Yes, this is all handled by the public method. `ReparsePoint` check should not be needed.

Reply:
updated.